src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1485,10 +1628,18 @@ private void ImportLoadString(int token)
 
         private void ImportInitObj(int token)
         {
+            TypeDesc type = ResolveTypeToken(token);
+            var valueEntry = _stack.Pop();
+            var llvmType = GetLLVMTypeForTypeDesc(type);
+            if (llvmType.TypeKind == LLVMTypeKind.LLVMArrayTypeKind)
+                LLVM.BuildStore(_builder, WebAssemblyObjectWriter.GetConstZeroArray(type.GetElementSize().AsInt), valueEntry.ValueAsType(LLVM.PointerType(llvmType, 0), _builder));
+            else
+                LLVM.BuildStore(_builder, BuildConstInt32(0), valueEntry.ValueAsType(LLVM.PointerType(LLVM.Int32Type(), 0), _builder));
```

Comment:
It looks like the comment about assuming this is an int or pointer wasn't addressed. What happens for a double* for example?